### PR TITLE
First go at a PR template for tools-iuc

### PR DESCRIPTION
**Pull request by Slugger70**
From the Tools IUC meeting at GCC this year, I was tasked with working on a review checklist. I thought that having a PR checklist template similar to the bioconda-recipes repo would be useful here as well. This is a WIP and please comment!

### CODE REVIEW

**Comment by shiltemann** [comment]
@Slugger70 cool :) 
@yhoogstrate once made a PR templater for our own galaxy tool repo that may give some more ideas? https://github.com/ErasmusMC-Bioinformatics/galaxytools-emc/blob/master/.github/PULL_REQUEST_TEMPLATE.md

**Comment by Slugger70** [comment]
Cool! Thanks @shiltemann. I'm also writing a comprehensive review checklist/guide for reviewers. But I reckon we should include something like @yhoogstrate's one here as well for reviewers too!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+* [ ] - This tool is appropriate for the tools-iuc repo (see [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md))
+* [ ] - This PR adds a new tool or tool collection
+* [ ] - This PR updates an existing tool or tool collection
+* [ ] - This PR does something else (explain below)
+
+FOR REVIEWER:
+* [ ] .shed.yml file ok
+* [ ] Indentation is correct
+* [ ] Tool version/build ok
+* [ ] `<command/>`: file names are escaped
```
Comment (mblue9) [inline .github/PULL_REQUEST_TEMPLATE.md]: > `<command/>`: file names are escaped

does this mean the element_identifier? e.g as in https://github.com/galaxyproject/tools-iuc/issues/1589  $element_identifier if used is escaped (see b47bba4)"

what about also something for single quotes like "Data and text params are single-quoted"?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+* [ ] Tests
+  - [ ] Parameters are reasonably covered
+  - [ ] Test files are appropriate
```
Comment (mblue9) [inline .github/PULL_REQUEST_TEMPLATE.md]: Is there any rule (of thumb) for what are appropriate test files? Does the total need to be < few MB?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+* [ ] Toolshed user `tools-iuc` has access to associated toolshed repo(s)
+* [ ] Complies with other best practice
```
Comment (mblue9) [inline .github/PULL_REQUEST_TEMPLATE.md]: Maybe add the link? http://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+FOR CONTRIBUTOR:
+* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document
+* [ ] - This tool is appropriate for the tools-iuc repo (see [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md))
+* [ ] - This PR adds a new tool or tool collection
+* [ ] - This PR updates an existing tool or tool collection
+* [ ] - This PR does something else (explain below)
+
+FOR REVIEWER:
+* [ ] .shed.yml file ok
+* [ ] Indentation is correct
```
Comment (mblue9) [inline .github/PULL_REQUEST_TEMPLATE.md]: Should the indentation always be 4 spaces?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (Slugger70) [reply .github/PULL_REQUEST_TEMPLATE.md]: Good thinking

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (Slugger70) [reply .github/PULL_REQUEST_TEMPLATE.md]: Yes

**Comment by yhoogstrate** [comment]
I am not sure if it is worthwile adding it to the list, but maybe it's worth considering:

 * [ ] It is legally allowed to use the tool without restrictions (unless mentioned in the PR)

**Comment by Slugger70** [comment]
@yhoogstrate I think so.. Not sure about how to word it. @erasche what do you think?

**Comment by erasche** [comment]
@Slugger70 yeah that's a good idea. We ask that in the cargo-port repo, [really great contributors](https://github.com/galaxyproject/cargo-port/pull/139) actually link the license as s requested, but most don't which annoys me. Maybe something like:

- [ ] License permits unrestricted use (educational + commercial)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+* [ ] .shed.yml file ok
+* [ ] Indentation is correct (4 spaces)
+* [ ] Tool version/build ok
+* [ ] `<command/>`
+  - [ ] Text parameters, input and output files `'single quoted'`
+  - [ ] Use of `<![CDATA[ ... ]]>` tags
+* [ ] Tests
+  - [ ] Parameters are reasonably covered
+  - [ ] Test files are appropriate
+* [ ] Toolshed user `tools-iuc` has access to associated toolshed repo(s)
```
Comment (shiltemann) [inline .github/PULL_REQUEST_TEMPLATE.md]: the toolshed user is just `iuc`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+FOR CONTRIBUTOR:
+* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document
+* [ ] - This tool is appropriate for the tools-iuc repo (see [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md))
```
Comment (nsoranzo) [inline .github/PULL_REQUEST_TEMPLATE.md]: Maybe merge this 2 in:
```
I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+* [ ] - This PR updates an existing tool or tool collection
+* [ ] - This PR does something else (explain below)
+
+FOR REVIEWER:
+* [ ] .shed.yml file ok
+* [ ] Indentation is correct (4 spaces)
+* [ ] Tool version/build ok
+* [ ] `<command/>`
+  - [ ] Text parameters, input and output files `'single quoted'`
+  - [ ] Use of `<![CDATA[ ... ]]>` tags
```
Comment (nsoranzo) [inline .github/PULL_REQUEST_TEMPLATE.md]: `<![CDATA[ ... ]]>` is recommended also for the `<help/>` section.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+* [ ] Tests
+  - [ ] Parameters are reasonably covered
+  - [ ] Test files are appropriate
+* [ ] Toolshed user `iuc` has access to associated toolshed repo(s)
```
Comment (nsoranzo) [inline .github/PULL_REQUEST_TEMPLATE.md]: This can go as sub-point of `.shed.yml file ok`.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+* [ ] - This PR adds a new tool or tool collection
+* [ ] - This PR updates an existing tool or tool collection
+* [ ] - This PR does something else (explain below)
+
+FOR REVIEWER:
+* [ ] .shed.yml file ok
+* [ ] Indentation is correct (4 spaces)
+* [ ] Tool version/build ok
+* [ ] `<command/>`
+  - [ ] Text parameters, input and output files `'single quoted'`
```
Comment (nsoranzo) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'd add:
```
  - [ ] Parameters of type `text` or having `optional="true"` attribute are checked with `if str($param)` before being used
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (nsoranzo) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'd also add
```
* [ ] Data parameters have a `format` attribute containing datatypes recognised by Galaxy

Comment (shiltemann) [reply .github/PULL_REQUEST_TEMPLATE.md]: maybe just add a checklist item that the help text is valid restructured text? (and uses CDATA tags)

**Comment by bgruening** [comment]
:+1:

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (nsoranzo) [reply .github/PULL_REQUEST_TEMPLATE.md]: @Slugger70 I was wondering if we should merge this list in the one you are adding in https://github.com/galaxyproject/tools-iuc/pull/2000 , and replace it here with a link to the guide for reviewers?

**Pull request merged**